pkg/runtime/namespace_config_provider.go

```diff
@@ -0,0 +1,41 @@
+package runtime
+
+import (
+	"context"
+
+	"github.com/lyft/flyteadmin/pkg/common"
+	"github.com/lyft/flyteadmin/pkg/runtime/interfaces"
+	"github.com/lyft/flytestdlib/config"
+	"github.com/lyft/flytestdlib/logger"
+)
+
+const (
+	namespaceMappingKey = "namespace_mapping"
+	domainVariable      = "domain"
+)
+
+var namespaceMappingConfig = config.MustRegisterSection(namespaceMappingKey, &interfaces.NamespaceMappingConfig{})
+
+type NamespaceMappingConfigurationProvider struct{}
+
+func (p *NamespaceMappingConfigurationProvider) GetNamespaceMappingConfig() common.NamespaceMapping {
+	var mapping string
+	if namespaceMappingConfig != nil && namespaceMappingConfig.GetConfig() != nil {
+		mapping = namespaceMappingConfig.GetConfig().(*interfaces.NamespaceMappingConfig).Mapping
+	}
+
+	switch mapping {
```

Comment:
can we also add an option for project-domain too that doesn't log as an unsupported option?

Reply:
added project-domain option